### ISSUE TITLE
configure smtp server

### ISSUE THREAD

**Issue by Tony-spearhead** [jjethwa/rundeck#83]
How do i pass these 2 variables to your docker image?
grails.mail.host=mymailserver.com
grails.mail.port=25
When I tried to maps these 2 values with the rundeck-config.properties rundeck always ended up in 503 service unavailable.

**Comment by jjethwa** [comment]
Hi @Tony-spearhead 

That should work if you've mounted /etc/rundeck as a volume.  Is there anything interesting in the rundeck logs under /var/log/rundeck ?

**Comment by Tony-spearhead** [comment]
No. I just get a 503 from jetty. I cannot mount /etc/rundeck at all

**Comment by Tony-spearhead** [comment]
rundeck:
    image: jordan/rundeck:latest
    container_name: "rundeck-infra"
    environment:
      SERVER_URL: http://myserver:4440
      RUNDECK_UID: rundeck
#     RDECK_GRAILS_MAIL_HOST: webcache.spearhead-ag.ch
#     RDECK_GRAILS_MAIL_PORT: 25
      RUNDECK_STORAGE_PROVIDER: db
      RUNDECK_PROJECT_STORAGE_TYPE: db
      EXTERNAL_SERVER_URL: http://myserver
      RDECK_JVM_SETTINGS: -Dfile.encoding=UTF-8
    volumes:
#     - /mnt/data/rundeck/etc/rundeck-groovy.config:/etc/rundeck/rundeck-groovy.config
      - /mnt/data/rundeck/logs:/var/log/rundeck
      - /mnt/data/rundeck/resources/resources.xml:/var/lib/rundeck/data/resources.xml
      - /mnt/data/rundeck/.ssh:/var/lib/rundeck/.ssh
      - /mnt/data/rundeck/etc/realm.properties:/etc/rundeck/realm.properties
      - /mnt/data/mysql:/var/lib/mysql
    network_mode: host
    logging:
      driver: gelf
      options:
        gelf-address: udp://mylog:12201
        tag: rundeck-infra

**Comment by jjethwa** [comment]
Hi @Tony-spearhead 

I don't see /etc/rundeck listed, just /mnt/data/rundeck/etc/realm.properties:/etc/rundeck/realm.properties

Try starting the container with an empty /etc/rundeck volume and it should create the default set of files for you.  You should be able to edit the files under the /etc/rundeck volume without issue

**Comment by Tony-spearhead** [comment]
Hello,

I will try that and keep you updated

Thanks for the help

**Comment by Tony-spearhead** [comment]
Hi @jjethwa 

I did that but the image does not create any files in the mounted etc directory.

Regards

**Comment by jjethwa** [comment]
Hi @Tony-spearhead 

The /etc/rundeck volume needs to be completely empty or else the default configs will not be created.  If the directory is empty and the files are not being created, can you check the docker logs to see if there are any complaints about permissions to the volume?

**Comment by eugeneduvenage** [comment]
I've created a pull request [https://github.com/jjethwa/rundeck/pull/86](https://github.com/jjethwa/rundeck/pull/86) that allows SMTP settings to be configured via environment variables.

**Comment by jjethwa** [comment]
Thanks @eugeneduvenage 😄 

@Tony-spearhead the changes will be in the latest tag in the next few minutes.  Please let us know if it works for you and close the issue if fixed.
